File: lab4/GCD.java
/************************************************************ 
 * * * GCD.java
 * * * Tangni Wang
 * * * twang63
 * * * Programming Assignment 4
 * * * This program calculates the GCD of two numbers.  
 * * ************************************************************/

import java.util.Scanner;

class GCD{
	public static void main( String[] args){
		Scanner sc = new Scanner(System.in);
		int x,y,bigN,smallN, r;
		System.out.print("Enter a positive integer: ");
		while(true){
			while(!sc.hasNextInt()){
				sc.next();
				System.out.print("Please enter a positive integer: ");
			}
			x = sc.nextInt();
			if( x>0 ) break;
			System.out.print("Please enter a positive integer: ");
		}System.out.print("Enter another positive integer: ");
		while(true){
                        while(!sc.hasNextInt()){
				sc.next();
				System.out.print("Please enter a positive integer: ");
			}
                        y = sc.nextInt();
                        if( y>0 ) break;
                        System.out.print("Please enter a positive integer: ");
                }

		if(x>y){
			bigN = x;smallN = y;
		}else{
			bigN = y;smallN = x;
		}
		r = 1;
		while(r!=0){
			r = bigN%smallN;
                	bigN = smallN;
                	smallN = r;
		}
		System.out.println("The GCD of "+ x + " and " + y + " is " + bigN);
	}
}

File: lab8/GCD.c
/************************************************************ 
 * GCD.c
 * Tangni Wang
 * twang63
 * Lab Assignment 8
 * This program calculates the GCD of two numbers.  
 *************************************************************/
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

int main(){
	int x, y, bigN, smallN, r, tempn;
	char temps[100001];
	char temps2[100001];
	printf("Enter a positive integer: ");
	int nread = scanf("%d",&x);
	while(nread == 0 || x<=0){
		printf("Please enter a positive integer: ");
		scanf("%s", temps);
		tempn = nread;
		if (tempn == 1) printf("Please enter a positive integer: ");
		nread = scanf("%d", &x);
		tempn = 0;
	}
	printf("Enter another positive integer: ");
	int nread2 = scanf("%d",&y);
        while(nread2 == 0 || y<=0){
                printf("Please enter a positive integer: ");
                scanf("%s", temps2);
		tempn = nread2;
		if (tempn == 1) printf("Please enter a positive integer: ");
		nread2 = scanf("%d", &y);
		tempn = 0;
        }
	if(x>y){
		bigN = x;smallN = y;
	}else{
		bigN = y; smallN = x;
	}r =1;
	while(r!=0){
		r = bigN%smallN;
		bigN = smallN;
		smallN = r;
	}printf("The GCD of %d and %d is %d\n", x, y, bigN);
	return 0;
}
